tests/abc_atlas_cache/test_cache.py

```diff
@@ -0,0 +1,573 @@
+        self.client.put_object(Bucket=self.test_bucket_name,
+                               Key=data_path,
+                               Body=data)
+
+        self.client.put_object(Bucket=self.test_bucket_name,
+                               Key=self.new_manifest_string,
+                               Body=bytes(json.dumps(manifest), 'utf-8'))
+
+        cache = S3CloudCache(self.cache_dir, self.test_bucket_name)
+
+        cache.load_manifest(self.new_manifest_string)
+
+        expected_path = (self.cache_dir / data_path)
+        assert not expected_path.exists()
+
+        # test data_path
+        directory = data_path.split('/')[1]
+        attr = cache.data_path(
+            directory=directory,
+            file_name=f"{data_path.split('/')[-1].split('.')[0]}/log2"
+        )
+        assert attr['local_path'] == expected_path
+        assert not attr['exists']
+
+        expected_path = self.cache_dir / metadata_path
+        assert not expected_path.exists()
+
+        directory = metadata_path.split('/')[1]
+        attr = cache.metadata_path(
+            directory,
+            metadata_path.split('/')[-1].split('.')[0]
+        )
+        assert attr['local_path'] == expected_path
+        assert not attr['exists']
+
+    def test_latest_manifest(self):
+        """
+        Test that the methods which return the latest and latest downloaded
+        manifest file names work correctly
+        """
+        manifest_list = self.create_manifests(manifest_count=5)
+
+        cache = S3CloudCache(self.cache_dir, self.test_bucket_name)
+
+        assert cache.latest_downloaded_manifest_file == ''
+
+        cache.load_manifest(manifest_list[2])
+        cache.load_manifest(manifest_list[0])
+        cache.load_manifest(manifest_list[1])
+
+        assert cache.latest_manifest_file == manifest_list[-1]
+
+        expected = manifest_list[2]
+        assert cache.latest_downloaded_manifest_file == expected
+
+    def test_outdated_manifest_warning(self):
+        """
+        Test that a warning is raised the first time you try to load an
+        outdated manifest
+        """
+        manifest_list = self.create_manifests(manifest_count=5)
+
+        cache = S3CloudCache(self.cache_dir, self.test_bucket_name)
+
+        m_warn_type = 'OutdatedManifestWarning'
+
+        with pytest.warns(OutdatedManifestWarning) as warn:
+            cache.load_manifest(manifest_list[0])
+        ct = 0
+        for w in warn.list:
+            if w._category_name == m_warn_type:
+                msg = str(w.message)
+                assert 'is not the most up to date' in msg
+                assert 'S3CloudCache.compare_manifests' in msg
+                assert 'load_latest_manifest' in msg
+                ct += 1
+        assert ct > 0
+
+        # assert no warning is raised the second time by catching
+        # any warnings that are emitted and making sure they are
+        # not OutdatedManifestWarnings
```

Comment:
this comment is no longer accurate (I think *any* warning will cause the test to fail).